zerver/lib/actions.py

```diff
@@ -4032,38 +4036,19 @@ def do_revoke_user_invite(invite_id: int, realm_id: int) -> None:
     prereg_user.delete()
     clear_scheduled_invitation_emails(email)
 
-def do_resend_user_invite_email(invite_id: int, realm_id: int) -> str:
-    try:
-        prereg_user = PreregistrationUser.objects.get(id=invite_id)
-    except PreregistrationUser.DoesNotExist:
-        raise JsonableError(_("Invalid invitation ID."))
-
-    if (prereg_user.referred_by.realm_id != realm_id):
-        raise JsonableError(_("Invalid invitation ID."))
+def do_resend_user_invite_email(prereg_user: PreregistrationUser) -> str:
+    check_invite_limit(prereg_user.referred_by, 1)
 
     prereg_user.invited_at = timezone_now()
     prereg_user.save()
 
-    # sends a invitation reminder since 'custom_body' can not be resent
-    # imported here to avoid import cycle error
-    from zerver.context_processors import common_context
-    clear_scheduled_invitation_emails(prereg_user.email)
+    do_increment_logging_stat(prereg_user.realm, COUNT_STATS['invites_sent::day'],
+                              None, prereg_user.invited_at)
 
-    link = create_confirmation_link(prereg_user,
-                                    prereg_user.referred_by.realm.host,
-                                    Confirmation.INVITATION)
-    context = common_context(prereg_user.referred_by)
-    context.update({
-        'activate_url': link,
-        'referrer_name': prereg_user.referred_by.full_name,
-        'referrer_email': prereg_user.referred_by.email,
-        'referrer_realm_name': prereg_user.referred_by.realm.name,
-    })
-    send_email(
-        "zerver/emails/invitation_reminder",
-        to_email=prereg_user.email,
-        from_address=FromAddress.NOREPLY,
-        context=context)
+    clear_scheduled_invitation_emails(prereg_user.email)
+    # We don't store the custom email body, so just set it to None
+    event = {"prereg_id": prereg_user.id, "referrer_id": prereg_user.referred_by.id, "email_body": None}
+    queue_json_publish("invites", event)
```

Comment:
I think the reminder email is better than the plain invite -- the reminder at least draws the connection that there was a previous message and maybe there's context there, while the fresh invite will have look kind of out of the blue.

But I'm not sure the reminder is quite the right text either. And definitely it's good to deduplicate this code. It might be a good non-urgent followup to go back through the text we use for these emails, and maybe make a new template (or variation of the reminder template) for these and thread the information through.

Also, it looks like in the old code we were actually sending the email from the web server process! Definitely seems right to move that work to the queue worker.